$NetBSD: patch-media_libcubeb_src_moz.build,v 1.1 2016/04/27 16:36:50 ryoon Exp $

Don't assume cocoa toolkit just because OS_TARGET is Darwin.

--- media/libcubeb/src/moz.build.orig	2016-02-25 23:01:58.000000000 +0000
+++ media/libcubeb/src/moz.build
@@ -17,6 +17,12 @@ if CONFIG['MOZ_ALSA']:
     ]
     DEFINES['USE_ALSA'] = True
 
+if CONFIG['MOZ_OSS']:
+    SOURCES += [
+        'cubeb_oss.c',
+    ]
+    DEFINES['USE_OSS'] = True
+
 if CONFIG['MOZ_PULSEAUDIO']:
     SOURCES += [
         'cubeb_pulse.c',
@@ -31,7 +37,7 @@ if CONFIG['OS_ARCH'] == 'OpenBSD':
     ]
     DEFINES['USE_SNDIO'] = True
 
-if CONFIG['OS_TARGET'] == 'Darwin':
+if CONFIG['MOZ_WIDGET_TOOLKIT'] == 'cocoa':
     SOURCES += [
         'cubeb_audiounit.c',
     ]
@@ -64,6 +70,9 @@ if CONFIG['OS_TARGET'] == 'Android':
 
 FINAL_LIBRARY = 'gkmedias'
 
+if CONFIG['MOZ_NATIVE_SPEEX']:
+    SOURCES['cubeb_resampler.cpp'].flags += CONFIG['MOZ_SPEEX_CFLAGS']
+
 if CONFIG['MOZ_WIDGET_TOOLKIT'] == 'gonk':
     if CONFIG['ANDROID_VERSION'] >= '17':
         LOCAL_INCLUDES += [
@@ -74,5 +83,6 @@ if CONFIG['MOZ_WIDGET_TOOLKIT'] == 'gonk
             '%' + '%s/system/media/wilhelm/include' % CONFIG['ANDROID_SOURCE'],
         ]
 
+CFLAGS += CONFIG['MOZ_OSS_CFLAGS']
 CFLAGS += CONFIG['MOZ_ALSA_CFLAGS']
 CFLAGS += CONFIG['MOZ_PULSEAUDIO_CFLAGS']
